pkgs/desktops/gnome/apps/gnome-nettool/default.nix

```diff
@@ -1,36 +1,71 @@
-{ lib, stdenv, fetchurl, pkg-config, gnome, gtk3, wrapGAppsHook
-, libgtop, intltool, itstool, libxml2, nmap, inetutils }:
+{ stdenv
+, lib
+, fetchurl
+, fetchpatch
+, desktop-file-utils
+, itstool
+, meson
+, ninja
+, pkg-config
+, python3
+, wrapGAppsHook
+, glib
+, gtk3
+, libgtop
+, gnome
+}:
 
 stdenv.mkDerivation rec {
   pname = "gnome-nettool";
-  version = "3.8.1";
+  version = "42.0";
 
   src = fetchurl {
-    url = "mirror://gnome/sources/${pname}/${lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
-    sha256 = "1c9cvzvyqgfwa5zzyvp7118pkclji62fkbb33g4y9sp5kw6m397h";
+    url = "mirror://gnome/sources/${pname}/${lib.versions.major version}/${pname}-${version}.tar.xz";
+    sha256 = "pU8p7vIDiu5pVRyLGcpPdY5eueIJCkvGtWM9/wGIdR8=";
   };
 
-  nativeBuildInputs = [ pkg-config ];
+  patches = [
+    # Fix build with meson 0.61
+    # https://gitlab.gnome.org/GNOME/gnome-nettool/-/merge_requests/3
+    (fetchpatch {
+      url = "https://gitlab.gnome.org/GNOME/gnome-nettool/-/commit/1124c3e1fdb8472d30b7636500229aa16cdc1244.patch";
+      sha256 = "fbpfL8Xb1GsadpQzAdmu8FSPs++bsGCVdcwnzQWttGY=";
+    })
+  ];
+
+  nativeBuildInputs = [
+    desktop-file-utils
+    itstool
+    meson
+    ninja
+    pkg-config
+    python3
+    wrapGAppsHook
+  ];
+
   buildInputs = [
-    gtk3 wrapGAppsHook libgtop intltool itstool libxml2
-    gnome.adwaita-icon-theme
+    glib
+    gtk3
+    libgtop
   ];
 
-  propagatedUserEnvPkgs = [ nmap inetutils ];
```

Comment:
nmap and inetutils still seem to be used in the source code:

https://gitlab.gnome.org/GNOME/gnome-nettool/-/blob/5748c0fa632c9caad9e46377ca1aa330dec4b757/src/scan.c#L77
https://gitlab.gnome.org/GNOME/gnome-nettool/-/blob/5748c0fa632c9caad9e46377ca1aa330dec4b757/src/traceroute.c#L97

Maybe add them to PATH via a wrapper?

Reply:
By checking the diff for `src/` I think actually no tools are added/deleted in the release.

By `grep -nr 'util_find_program'` I saw:

```
ChangeLog:462:	* src/utils.c (util_find_program_dialog): Fixed #154069
ChangeLog:1561:	* utils.c (util_find_program_dialog): Verify if the parent
ChangeLog:1709:		util_find_program_dialog, util_tree_model_to_string,
ChangeLog:1710:		util_find_program_in_path, my_strchrnul.
src/callbacks.c:147:		network_tool_path = util_find_program_in_path ("nm-connection-editor", NULL);
src/callbacks.c:151:			network_tool_path = util_find_program_in_path ("network-admin", NULL);
src/finger.c:83:	program = util_find_program_in_path ("finger", NULL);
src/finger.c:86:		program = util_find_program_dialog ("pinky", NULL);
src/info.c:335:	network_tool_path = util_find_program_in_path ("nm-connection-editor", NULL);
src/info.c:337:		network_tool_path = util_find_program_in_path ("network-admin", NULL);
src/lookup.c:101:	program = util_find_program_dialog ("dig", parent);
src/netstat.c:199:	program = util_find_program_dialog ("netstat", parent);
src/ping.c:301:		program = util_find_program_dialog ("ping", parent);
src/ping.c:304:		program = util_find_program_dialog ("ping6", parent);
src/scan.c:77:	program = util_find_program_dialog ("nmap", parent);
src/traceroute.c:79:		program = util_find_program_in_path ("tracepath", NULL);
src/traceroute.c:91:		program = util_find_program_in_path ("tcptraceroute", NULL);
src/traceroute.c:96:			program = util_find_program_dialog ("traceroute", parent);
src/traceroute.c:102:		program = util_find_program_in_path ("tracepath6", NULL);
src/traceroute.c:108:			program = util_find_program_dialog ("traceroute6", parent);
src/utils.c:72:util_find_program_in_path (const gchar * program, const gchar * path)
src/utils.c:220:util_find_program_dialog (gchar * program, GtkWidget *parent) {
src/utils.c:229:	result = util_find_program_in_path (program, my_path);
src/utils.h:38:gchar * util_find_program_in_path (const gchar * program, const gchar * path);
src/utils.h:39:gchar * util_find_program_dialog (gchar * program, GtkWidget *parent);
src/whois.c:76:	program = util_find_program_dialog ("whois", parent);
```

I am actually not sure if I should add all of them to PATH, and I am also not sure which pacakge I should use (for example ping6 and tracepath6 is removed in iputils in https://github.com/NixOS/nixpkgs/pull/22822 (year 2017?))

Reply:
I can still see the 6 commands in netutils.

Reply:
(Sorry I am not able to test IPv6 related stuff currently :crying_cat_face:) Is inetutils a good provider for e.g. `ping6`? For example by using `ping` provided by inetutils (can be tested by changing `--suffix PATH` to `--prefix PATH`), I got:

```
ping: invalid option -- 'b'
Try 'ping --help' or 'ping --usage' for more information.
```
<img src="https://user-images.githubusercontent.com/20080233/162252061-b95153ff-fc67-4f67-88d7-276519ed6836.png" width=50%>

Reply:
Yeah. according to https://unix.stackexchange.com/a/400354/41260 inetutils is more limited. So we probably want `--prefix` with `iputils nmap inetutils` (or maybe `traceroute` before `inetutils` if the program also expects that version). Either way, this is probably a bug that should be reported upstream.